src/sc/sidechainTxsCommitmentGuard.cpp

```diff
@@ -0,0 +1,280 @@
+bool SidechainTxsCommitmentGuard::add(const CScCertificate& cert) { return true; }
+#else
+
+bool SidechainTxsCommitmentGuard::add_fwt(const CTxForwardTransferOut& ccout)
+{
+    LogPrint("sc", "%s():%d entering \n", __func__, __LINE__);
+
+    // Check against the number of sidechains currently in the commitment tree
+    // If we are adding a new sc and we already hit the limit, do not add anything
+    if (!cbs.checkAvailableSpaceAliveSC(ccout.GetScId()))
+        return false;
+
+    // The sidechain id should not be already in the ceased tree
+    if (cbs.checkExistenceInCeasedSCTree(ccout.GetScId())) {
+        LogPrint("sc", "%s():%d - scTxsCommitment guard failed: adding fwt for scId[%s] on alive subtree, but already added on ceased subtree.\n",
+            __func__, __LINE__, ccout.GetScId().ToString());
+        return false;
+    }
+
+    CommitmentBuilderStatsAliveCounter& counterRef = cbs.cbsaMap[ccout.GetScId()];
+
+    // Check against the number of FT already present in the sc subtree.
+    // Only try to add if under the limit; increase the counter only if add was successful,
+    // as adding may fail for reasons different than having a full subtree.
+    if (counterRef.ft < cbs.FT_LIMIT) {
+        ++counterRef.ft;
+        LogPrint("sc", "%s():%d - scTxsCommitment guard: successfully added FT for sidechain scId[%s] - current size: %d.\n",
+            __func__, __LINE__, ccout.GetScId().ToString(), counterRef.ft);
+        return true;
+    } else {
+        LogPrint("sc", "%s():%d - scTxsCommitment guard failed: too many FT for sidechain scId[%s].\n",
+            __func__, __LINE__, ccout.GetScId().ToString());
+        return false;
+    }
+}
+
+bool SidechainTxsCommitmentGuard::add_bwtr(const CBwtRequestOut& ccout)
+{
+    LogPrint("sc", "%s():%d entering \n", __func__, __LINE__);
+
+    // Check against the number of sidechains currently in the commitment tree
+    // If we are adding a new sc and we already hit the limit, do not add anything
+    if (!cbs.checkAvailableSpaceAliveSC(ccout.GetScId()))
+        return false;
+
+    // The sidechain id should not be already in the ceased tree
+    if (cbs.checkExistenceInCeasedSCTree(ccout.GetScId())) {
+        LogPrint("sc", "%s():%d - scTxsCommitment guard failed: adding bwtr for scId[%s] on alive subtree, but already added on ceased subtree.\n",
+            __func__, __LINE__, ccout.GetScId().ToString());
+        return false;
+    }
+
+    CommitmentBuilderStatsAliveCounter& counterRef = cbs.cbsaMap[ccout.GetScId()];
+
+    // Check against the number of BWT already present in the sc subtree.
+    // Only try to add if under the limit; increase the counter only if add was successful,
+    // as adding may fail for reasons different than having a full subtree.
+    if (counterRef.bwtr < cbs.BWTR_LIMIT) {
+        ++counterRef.bwtr;
+        LogPrint("sc", "%s():%d - scTxsCommitment guard: successfully added BWTR for sidechain scId[%s].\n",
```

Comment:
(very minor importance) Maybe this info can be added  to the log: `- current size: %d.\n",`; at least to get aligned with `add_fwt`. Similar comment applies for `add_csw` and `add_cert`.

Reply:
Thanks, ass suggested by @ptagl  we removed all the logs concerning successful adds to the maps